### ISSUE TITLE
Xdebug not working

### ISSUE THREAD

**Issue by arrubiu** [wodby/docker4drupal#400]
I'm using last release of docker4drupal on a linux host and I can't use xdebug.
I try to add a breakpoint, the server seems slow, but it doesn't stop on breakpoint.

What's wrong?

I've a Drupal 8 installation based on composer, so the directory structure is:

- document_root
  - drush
  - mariadb-init
  - scripts
  - vendor
  - web

On my docker-compose.yml, on php service, I've this:
```
  PHP_XDEBUG: 1
  PHP_XDEBUG_DEFAULT_ENABLE: 1
  PHP_XDEBUG_REMOTE_CONNECT_BACK: 0
  PHP_XDEBUG_REMOTE_LOG: /tmp/php-xdebug.log
```
Here is my PHPStorm configuration for debugging:
 
![Schermata del 2019-09-03 14-39-22](https://user-images.githubusercontent.com/1532246/64173935-aad46180-ce58-11e9-9be7-f9faacb80856.png)

Here is the output of php-xdebug.log:
```
root@php.container:/tmp $ cat php-xdebug.log 
[87] Log opened at 2019-09-03 12:44:57
[87] I: Connecting to configured address/port: localhost:9000.
[87] I: Connected to client. :-)
[87] -> <init xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" fileuri="file:///var/www/html/web/index.php" language="PHP" xdebug:language_version="7.3.8" protocol_version="1.0" appid="87" idekey="www-data"><engine version="2.7.2"><![CDATA[Xdebug]]></engine><author><![CDATA[Derick Rethans]]></author><url><![CDATA[https://xdebug.org]]></url><copyright><![CDATA[Copyright (c) 2002-2019 by Derick Rethans]]></copyright></init>
[87]
[87] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" status="stopping" reason="ok"></response>
[87]
[87] Log closed at 2019-09-03 12:45:10
[87]
[86] Log opened at 2019-09-03 12:45:10
[86] I: Connecting to configured address/port: localhost:9000.
[86] I: Connected to client. :-)
[86] -> <init xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" fileuri="file:///var/www/html/web/core/modules/statistics/statistics.php" language="PHP" xdebug:language_version="7.3.8" protocol_version="1.0" appid="86" idekey="www-data"><engine version="2.7.2"><![CDATA[Xdebug]]></engine><author><![CDATA[Derick Rethans]]></author><url><![CDATA[https://xdebug.org]]></url><copyright><![CDATA[Copyright (c) 2002-2019 by Derick Rethans]]></copyright></init>
[86]
[86] -> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="https://xdebug.org/dbgp/xdebug" status="stopping" reason="ok"></response>
[86]
[86] Log closed at 2019-09-03 12:45:10
[86]
root@php.container:/tmp $ 
```

**Comment by spleshka** [comment]
I'm experiencing the same issue. Tbh, xdebug stoped working for my team a while ago. Our workaround for a while was using quite old images, but now we *need* latest images and therefore it became a real issue.

The relevant part of my config:

```
      PHP_XDEBUG: 1
      PHP_XDEBUG_DEFAULT_ENABLE: 1
      PHP_XDEBUG_REMOTE_AUTOSTART: 1
      PHP_XDEBUG_REMOTE_CONNECT_BACK: 0
      PHP_XDEBUG_REMOTE_HOST: host.docker.internal # Docker 18.03+ Mac/Win
```

The PHPStorm breaks at the first line and does not go any further than that. So it captures the xdebug connection but that's it. I'll try to investigate and come up with more details if I find anything.

**Comment by spleshka** [comment]
Actually, turns out the issue is in the version of PHPStorm used (it was 2018.1). I've upgraded to 2019.2 and everything works flawlessly :)

**Comment by arrubiu** [comment]
I've downloaded now the last version of PhpStorm, xdebug doesn't work.
I've added your configuration above:

```
PHP_XDEBUG: 1
PHP_XDEBUG_DEFAULT_ENABLE: 1
PHP_XDEBUG_REMOTE_AUTOSTART: 1
PHP_XDEBUG_REMOTE_CONNECT_BACK: 0
PHP_XDEBUG_REMOTE_HOST: host.docker.internal # Docker 18.03+ Mac/Win
```

**Comment by Niklan** [comment]
As I remember, xdebug still not working with PHP 7.3+. So you must use PHP 7.2- to get xdebug working.

**Comment by spleshka** [comment]
I do use XDebug with php 7.3 and it works fine. It used to be an issue a while ago, but no longer the case with latest versions.

**Comment by arrubiu** [comment]
Nothing :(
I've tried also 7.3-4.13.10 and 7.2-4.13.10

**Comment by csandanov** [comment]
> As I remember, xdebug still not working with PHP 7.3+. So you must use PHP 7.2- to get xdebug working.

Xdebug supports 7.3 since 4.11.3 https://github.com/wodby/php/issues/56

**Comment by Niklan** [comment]
Indeed, I don't even notice that my project already working on 7.3 and I actively use xdebug :) Just remembered, previously if xdebug not respond, then I need to downgrade PHP.

**Comment by arrubiu** [comment]
I solved using PhpStorm 2019.2 and using the button "Start listening for php debug connection" instead of configuring the debugger as explained in the first post.
In this way works with php 7.3 and 7.2.
So I've to buy the new phpstorm..
